### PR TITLE
[MB-1790][MB-1778]Fix Subscription filtering and Refresh button

### PR DESCRIPTION
**Pull request by sirojan24**
Fix subscription filtering for durable topics and fix the refresh button.
Associated JIRA Link:  https://wso2.org/jira/browse/MB-1778
https://wso2.org/jira/browse/MB-1790

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java [hunk 1]
```diff
@@ -569,7 +569,7 @@ private static void addLocalSubscriptionsForAllBindingsOfQueue(AMQQueue queue, S
                     }
 
                     ProtocolType protocol = ProtocolType.AMQP;
-                    String subscriptionIdentifier = "";
+                    String subscriptionIdentifier = queue.getName();
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java]: What was the reason for doing this?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java [hunk 2]
```diff
@@ -114,6 +115,19 @@ public String getObjectInstanceName() {
     /**
      * {@inheritDoc}
      */
+    public long getPendingMessageCount(String subscriptionID) throws MBeanException {
+        try {
+            return MessagingEngine.getInstance().getMessageCountOfQueue(subscriptionID);
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java]: Does subscription ID always equal to queue name?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java (hunk 1 above)
Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java]: The subscription identifier is the storage queue name.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java (hunk 2 above)
Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java]: yes. from the UI we always send the queue name

File: modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java (hunk 1 above)
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java]: Isn't it better to rename this local variable to reflect that? or else we should at lease have a code comment explaining why we do this.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java (hunk 2 above)
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java]: Then better to change the parameter name.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java (hunk 1 above)
Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java]: Comment added

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java (hunk 2 above)
Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java]: Parameter name changed in the latest commit

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java [hunk 3]
```diff
@@ -47,6 +47,7 @@
     private static Log log = LogFactory.getLog(SubscriptionManagementInformationMBean.class);
 
     private static final String SEPARATOR = ";";
+    private static final String SELECT_ALL = "All";
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java]: let's add comments

File: modules/andes-core/management/common/src/main/java/org/wso2/andes/management/common/mbeans/SubscriptionManagementInformation.java [hunk 4]
```diff
@@ -46,7 +46,17 @@
             @MBeanOperationParameter(
                     name = "destinationType",
                     description = "Destination type of the subscriptions") String destinationType)
-    throws MBeanException;
+            throws MBeanException;
+
+    /**
+     * Get the pending message count for the specified subscription
+     * @param queueName for which the pending message cound need to be calculated
```
Comment (sdkottegoda) [inline modules/andes-core/management/common/src/main/java/org/wso2/andes/management/common/mbeans/SubscriptionManagementInformation.java]: Let's write this according to the java doc format. See http://www.oracle.com/technetwork/articles/java/index-137868.html

File: modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java (hunk 1 above)
Comment (sajinidesilva) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java]: Add the commit link here

Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java]: https://github.com/wso2/andes/pull/707

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java (hunk 3 above)
Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/SubscriptionManagementInformationMBean.java]: Comment added in https://github.com/wso2/andes/pull/714

File: modules/andes-core/management/common/src/main/java/org/wso2/andes/management/common/mbeans/SubscriptionManagementInformation.java (hunk 4 above)
Comment (sirojan24) [inline modules/andes-core/management/common/src/main/java/org/wso2/andes/management/common/mbeans/SubscriptionManagementInformation.java]: Done in https://github.com/wso2/andes/pull/714

**Pull request merged**